### PR TITLE
support resources with dots in their Kind field

### PR DESCRIPTION
**Pull request by akostadinov**
@xiuwang, please test whether this will help #308

you can refer to the name "cluster" of kind "config.samples.operator.openshift.io" by `config_samples_operator_openshift_io("cluster")`. And let me know!

@pruan-rht, notice file URL change here

### CODE REVIEW

**Comment by xiuwang** [comment]
Alex, Thanks.
I will test this pr.

File: lib/admin_credentials.rb [hunk 1]
```diff
@@ -68,12 +68,12 @@ def get
...
     end
   end
- 
+
   class URLKubeconfigCredentials < AdminCredentials
     def get
       # export OPENSHIFT_ENV_OCP4_ADMIN_CREDS_SPEC=file:///~/.kube/config
       if opts[:spec].start_with? 'file:///'
-        path = File.expand_path(opts[:spec].split("file:///")[1])
+        path = File.expand_path(opts[:spec].split("file://")[1])
```
Comment (pruan-rht) [inline lib/admin_credentials.rb]: @akostadinov any reason to change the local `file:`` directive?

Comment (pruan-rht) [reply lib/admin_credentials.rb]: https://en.wikipedia.org/wiki/File_URI_scheme

Comment (akostadinov) [reply lib/admin_credentials.rb]: Yes. According to that page, if you put a URL `file:///tmp/test.kubeconfig` then you expect an absolute PATH `/tmp/test.kubeconfig`. Without the change it is treated as the relative `tmp/test.kubeconfig` PATH.

If we want to allow relative paths, we need to also update line 75 to make it also allow `file://` (only to slashes) although this is not allowed by specification. WDYT?

Comment (pruan-rht) [reply lib/admin_credentials.rb]: @akostadinov I think we should stick with a scheme that's valid specification. I don't think relative paths are that big of a deal.

Comment (akostadinov) [reply lib/admin_credentials.rb]: I thought you needed them, that's why I suggested we can support such. In any case my change I believe is according to spec.

Comment (pruan-rht) [reply lib/admin_credentials.rb]: @akostadinov ok, I have no objections then.

Comment (pruan-rht) [reply lib/admin_credentials.rb]: LGTM, once @xiuwang has verified this PR works, then we can merge it.

Comment (xiuwang) [reply lib/admin_credentials.rb]: @akostadinov @pruan-rht Sorry for laying down the steps, the PR works, you can merge it now
Thanks～

**Pull request merged**